docker/Dockerfile

```diff
@@ -6,6 +6,7 @@ COPY environment.yml /tmp/environment.yml
 RUN conda env update -f /tmp/environment.yml -n base && rm /tmp/environment.yml
 
 COPY . /tmp/stactools-planet
+RUN pip install -r /tmp/stactools-planet/requirements-dev.txt
```

Comment:
Otherwise we try to pull in a GDAL source distribution that can't be built.

Reply:
I have another idea I want to try, which is getting everything from conda and then running `pip install . --no-deps`.

Reply:
The gist of it: pip, even when run in a conda environment, has no way of knowing that a GDAL conda package has already been installed.

Reply:
I decided not to change environment.yml. Pip-installing stactools into a conda environment is sorta broken and can't be nicely fixed until GDAL is gone. Bottom line: if a package is going to be pip installed (ie isn't in conda-forge) then all of its dependencies need to be on PyPI as well if the install experience is to be a good one.

Reply:
So there's a `Dockerfile-dev` that's used for the dev dependencies -- I _think_ we were trying to keep this Dockerfile to be a "execution only" container (i.e. as light as possible)?

Reply:
@gadomski I'm not sure of the intent of the docker stuff. But `docker/cibuild` doesn't succeed unless stactools is kept to <= 0.2.3 (because of gdal). I could create a new requirements.txt with just the minimum to use here.